Fix misc 'crc config' issues

cfergeau opened the pull request:
This PR fixes miscellaneous issues related to "crc config". The most important one is the first commit avoiding writing default values to the config file.
The others are either refactors to gather all configuration-related code in a single place, or cosmetic improvements to "crc config" output.

cfergeau commented on pkg/crc/config/viper_config.go:
```diff
@@ -26,22 +26,28 @@ func GetBool(key string) bool {
+func Unset(key string) error {
+	delete(ViperConfig, key)
+	/* This apparently preserves default values - unit test would probably be useful.. */
+	return syncViperState(globalViper)
```
This is what I want, and the comment is out of place now, I should remove it.
It was about doing globalViper.SetDefault() followed by globalViper.ReadConfig().
I believe the right thing is happening, but I did not check closely ;)

anjannath replied:
Yes, its doing the right thing now :)

praveenkumar commented on cmd/crc/cmd/config/config.go:
```diff
@@ -61,10 +42,44 @@ to the options that you set when you run the 'crc start' command.
+	return strings.HasPrefix(key, "skip-") || strings.HasPrefix(key, "warn-")
+}
+
+func less(lhsKey, rhsKey string) bool {
```
It really took me sometime to understand what this function do, better to put a comment on the function to make sure next time we don't spend much time to figure out what it actually do :)

praveenkumar commented on cmd/crc/cmd/config/get.go:
```diff
@@ -26,9 +27,9 @@ to the options that you set when you run the 'crc start' command.`,
+		errors.ExitWithMessage(1, err.Error())
 	}
-	output.Out("Config not found:", key)
+	output.Out(key, ":", v)
```
👍

praveenkumar commented on cmd/crc/cmd/config/set.go:
```diff
@@ -25,28 +25,12 @@ to the options that you set when you run the 'crc start' command.`,
-		errors.ExitWithMessage(1, "Config property does not exist: %s", key)
+	err := config.Set(key, value)
+	if err != nil {
+		errors.ExitWithMessage(1, err.Error())
```
That you could've done same `if err := config.Set(key, value); err !=nil` but that is not blocker.

cfergeau replied:
yes, codebase currently uses a mix of the 2. I'll change it so that it's at least consistent in the set.go file ;)

praveenkumar commented on pkg/crc/config/viper_config.go:
```diff
@@ -81,29 +96,54 @@ func InitViper() error {
 
+func SetDefaults() {
+	for _, setting := range allSettings {
+		setDefault(setting.Name, setting.defaultValue)
```
👍

gbraad commented:
A new push followed after approval? @cfergeau what did change ?

cfergeau commented:
> A new push followed after approval? @cfergeau what did change ?

I addressed the 2 comments raised by Praveen, changed `err := ...; if err {` to `if err := ..., err != nil {`, and added a comment to `less`

praveenkumar commented on cmd/crc/cmd/config/config.go:
```diff
@@ -61,10 +42,48 @@ to the options that you set when you run the 'crc start' command.
+// less is used to sort the config keys. We want to sort first the regular keys, and
+// then the keys related to preflight starting with a skip- or warn- prefix. We want
+// these preflight keys to be grouped by pair: 'skip-bar', 'warn-bar', 'skip-foo', 'warn-foo'
+// would be sorted in that order.
```
👍
